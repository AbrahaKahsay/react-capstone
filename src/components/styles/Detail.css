.coin-name {
    text-align: center;
    background-color: rgb(88,137,228);
}

.cont-det {
    margin: 0;
    background-color: rgb(67,105,178);
    color: white;
    min-height: 100vh;
    padding-top: 10px;
}

.back-icon-div{
    background-color: rgb(67,105,178);
}

.back-icon {
    height: 30px;
    width: 30px;
    display: flex;
    align-items: flex-start;
    align-self: center;
}