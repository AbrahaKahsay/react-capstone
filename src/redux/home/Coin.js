// API for coins
const baseUrl = 'https://api.coincap.io/v2/assets?limit=100';

// Actions
const COINS_FETCHED = 'COINS_FETCHED';
const COINS_FILTERED = 'COINS_FILTERED';

const initialState = [];

// Getting the coins from server
const fetchCoins = async () => {
    const get = await fetch(baseUrl, { method: 'GET' });
    const response = await get.json();
    const coins = response.data;
    return coins;
  };

export const coinList = () => async (dispatch) => {
    const coins = await fetchCoins();
    const firstCoins = coins.slice(0, 51);
    dispatch({
        type: COINS_FETCHED,
        newState: firstCoins,
    });
};

export const coinFilter = (filter) => async (dispatch) => {
    const coins = await fetchCoins();
    if (filter === 'name') {
      coins.sort((a, b) => a.name.localeCompare(b.name));
    } else if (filter === 'vol') {
      coins.sort((a, b) => b.priceUsd - a.priceUsd);
    }
    const firstCoins = coins.slice(0, 51);
    dispatch({
      type: COINS_FILTERED,
      filterState: firstCoins,
    });
  };

const coinReducer = (state = initialState, action) => {
    switch (action.type) {
        case FETCHCOINS:
            return action.newState;
        case FILTERCOINS:
            return action.filterState;
        default:
            return state;
    }
  };
  
  export default coinReducer;