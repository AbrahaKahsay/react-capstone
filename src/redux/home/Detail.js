const CoinAPI = 'https://api.coincap.io/v2/assets/';

const DETAIL_FETCHED = 'DETAIL_FETCHED';
const initialState = [{ id: 1, symbol: '$$' }];

const fetchDetail = async (id) => {
  const get = await fetch(apiCoin + id, { method: 'GET' });
  const coins = await get.json();
  return coins;
};

export const coinDetail = (id) => async (dispatch) => {
    const coinDet = await fetchDetail(id);
    dispatch({
      type: DETAIL_FETCHED,
      detail: coinDet.data,
    });
  };

  const detailReducer = (state = initialState, action) => {
    switch (action.type) {
      case DETAIL_FETCHED:
        return action.detail;
      default:
        return state;
    }
  };
  
  export default detailReducer;