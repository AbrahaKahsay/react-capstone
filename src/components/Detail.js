import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { NavLink, useParams } from 'react-router-dom';
import { coinDetail } from '../redux/home/Detail';
import Back from '../images/back.png';

const Detail = () => {
  const { id } = useParams();
  const dispatch = useDispatch();
  const coin = useSelector((store) => store.detailReducer);
  useEffect(() => {
    dispatch(coinDetail(id));
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <section>
      <NavLink to="/">
        <img className="icons" src={Back} alt="back" />
      </NavLink>

      <div className="detail">
        <h2>{coin.symbol}</h2>
        <p>{coin.name}</p>
      </div>
      <div className="title">
        Details
      </div>
      <div className="detailList">
        <div>
          <p>Rank</p>
        </div>
        <div>
          <p>
            {' #'}
            {coin.rank}
          </p>
        </div>
        <div className="d">
          <p>Price</p>
        </div>
        <div className="d">
          <p>
            {' $'}
            {coin.priceUsd}
            {' USD'}
          </p>
        </div>
        <div>
          <p>Volume 24h</p>
        </div>
        <div>
          <p>
            {' $'}
            {coin.volumeUsd24Hr}
          </p>
        </div>
        <div className="d">
          <p>Total Supply</p>
        </div>
        <div className="d">
          <p>
            {' '}
            {coin.supply}
            {' '}
            {coin.symbol}
          </p>
        </div>
      </div>
    </section>
  );
};
export default Detail;
