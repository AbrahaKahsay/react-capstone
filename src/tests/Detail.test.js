import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import Detail from '../components/Detail';
import { coinDetail } from '../redux/home/Detail';

describe('Jest Snapshot testing suite', () => {
  test('component match with the snapshot', async () => {
    await store.dispatch(coinDetail('bitcoin'));
    const tree = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Detail />
        </Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
