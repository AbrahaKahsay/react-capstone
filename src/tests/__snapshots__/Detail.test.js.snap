// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite Matches DOM Snapshot 1`] = `
<section
  class="cont-det"
>
  <div
    class="back-icon-div"
  >
    <a
      aria-current="page"
      class="active"
      href="/"
    >
      <img
        alt="back"
        class="back-icon"
        src="back.png"
      />
    </a>
  </div>
  <div
    class="detail coin-name"
  >
    <h2 />
    <p />
  </div>
  <div
    class="det-title col"
  >
    Details
  </div>
  <div
    class="detailList"
  >
    <div
      class="d"
    >
      <p>
        Rank
      </p>
      <div>
        <p>
           #
        </p>
      </div>
    </div>
    <div
      class="d col"
    >
      <p>
        Price
      </p>
      <div>
        <p>
           $
           USD
        </p>
      </div>
    </div>
    <div
      class="d"
    >
      <p>
        Volume 24h
      </p>
      <div>
        <p>
           $
        </p>
      </div>
    </div>
    <div
      class="d col"
    >
      <p>
        Total Supply
      </p>
      <div>
        <p>
           
           
        </p>
      </div>
    </div>
  </div>
</section>
`;
